### ISSUE TITLE
In-line csv files should use glue::trim

### ISSUE THREAD

**Issue by hadley** [tidyverse/readr#650]
So you can naturally indent strings, e.g.

```
read_csv("
  a,b,c
  1,2,3
")
```

**Comment by jimhester** [comment]
I agree this would be nice for the examples. However I could also see needing to opt-out of this in some cases, for instance testing edge cases, and I don't see a good way to do that, particularly that we now use `I()` for this, which doesn't take any arguments.
